From 1363802254f58d5244d1fa57edc8c9ee2d5cbd38 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Thu, 10 Jun 2021 13:23:18 +0200
Subject: [PATCH 4/5] alsa: use the local alibpref of the card

The _alibpref of the device was created in the session manager and
does not match our local _alibpref. Patch the device name with
the local _alibpref to make things match.

See #1286
---
 spa/plugins/alsa/alsa-pcm.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/spa/plugins/alsa/alsa-pcm.c b/spa/plugins/alsa/alsa-pcm.c
index 6b9625584..d7572da09 100644
--- a/spa/plugins/alsa/alsa-pcm.c
+++ b/spa/plugins/alsa/alsa-pcm.c
@@ -23,6 +23,7 @@ int spa_alsa_init(struct state *state)
 
 	if (state->open_ucm) {
 		char card_name[64];
+		const char *alibpref = NULL;
 
 		snprintf(card_name, sizeof(card_name), "hw:%i", state->card_index);
 		err = snd_use_case_mgr_open(&state->ucm, card_name);
@@ -44,6 +45,14 @@ int spa_alsa_init(struct state *state)
 				return err;
 			}
 		}
+
+		if ((snd_use_case_get(state->ucm, "_alibpref", &alibpref) != 0))
+			alibpref = NULL;
+		if (alibpref != NULL) {
+			size_t len = strlen(alibpref);
+			if (len > 4 && strncmp(state->props.device, alibpref, 4) == 0)
+				memcpy(state->props.device, alibpref, len);
+		}
 	}
 	return 0;
 }
-- 
2.31.1

